### DISCUSSION TITLE
How to use "updateFields: true" in existing table of content together with patchDocument?

### DISCUSSION THREAD

**Discussion by 0x1234567890** [dolanmiu/docx#2394] [Q&A]
Hello,

I want to edit an existing Word document using the patchDocument Patcher. My Word document already contains a table of contents, and I want the Patcher to update the existing table of contents when the document is output.

It seems there is an attribute called "updateFields: true" for this purpose. However, I'm not sure how to use this in conjunction with the Patcher, especially when there's already a table of contents in my document.

Currently, after modifying the Word file, I have to open the document and manually update the entire table of contents, which works. How can I incorporate this into the process?

My Code is simple:

```
patchDocument(fs.readFileSync("template.docx"), {
    patches: {
        headline: {
            type: PatchType.PARAGRAPH,
            children: [new TextRun("My New Headline")],
        },
    },
}).then((doc) => {
    fs.writeFileSync("output.docx", doc);
});

```

----

i also try to insert a new table of content but then the Word File do open with a warning to refresh the table of content:

```
patchDocument(fs.readFileSync("template.docx"), {
    patches: {
        headline: {
            type: PatchType.PARAGRAPH,
            children: [new TextRun("My New Headline")],
        },
		summary: {
            type: PatchType.DOCUMENT,
            children: [ new TableOfContents("summary", {
                    hyperlink: true,
                    headingStyleRange: "1-5",
                })],
        },
    },
}).then((doc) => {
    fs.writeFileSync("output.docx", doc);
});

```

How can i add the "updateFields: true"?

**Comment by dolanmiu** [comment]
This would be fairly difficult because Word does the rendering of the TOC

and then based on the render, it figures out what page numbet it is etc

Probably out of scope, closing

Related to: [updateFields](https://github.com/dolanmiu/docx/issues/1212)
